Topics/System_Design/Authentication_Service_Design.md

```diff
@@ -0,0 +1,85 @@
+2. [Sessions and Tokens](#sessions-tokens)
+	1. [Designing Token-Based Authentication](#token-design)
+4. [Design Considerations](#auth-design)
+5. [Common Technologies](#auth-common-tech)
+6. [Further Reading](#further-reading)
+
+## Types of Authentication <a name="auth-types"></a>
+
+We all know about usernames and passwords, or **single-factor authentication**, which grants access to a system when the credentials are correct. However, in most modern systems, this is no longer the case for several reasons:
+1. Users don't keep their passwords secure
+2. Users are vulnerable to phishing schemes
+3. Users easily forget their own credentials
+4. Credentials can be easily cracked
+
+A common theme we'll see is that authentication schemes will try to work around the weakest security link in the system, which are the users themselves. While small improvements can help, like security questions or increasing requirements for password hygiene (minimum N characters, with at least M special characters), modern approaches have better ways of working around this limitation. 
+
+### Multi-Factor Authentication (MFA) <a name="mfa"></a>
+
+Some MFA strategies require users to prove their credentials by demonstrating proof of *physical ownership* of an item. The most common among these is requiring the user to confirm their identity on a **separate device**, like a phone or a laptop; however, a really determined threat actor can bypass this by gaining remote access to the user's devices. 
+
+The best option of MFA strategies is by making the user prove their *inherent characteristics*; that is, things that are unique to them that no one else can ever take from them. **Biometrics** are a good example of this; many smartphones will use fingerprints and facial recognition, as an example.
+
+### Standardized Protocols <a name="protocols"></a>
+
+Developers don't have to write everything from scratch; using a open protocol ensures that a service can have a standardized yet secure authentication mechanism. Several protocols exist (definitions from [this article](https://www.getkisi.com/blog/authentication-protocols-overview)):
+1. **[LDAP (Lightweight Directory Access Protocol)](https://www.getkisi.com/glossary/lightweight-directory-access-protocol)** allows users to "securely locate organizations, individuals, and other resources such as files and devices in a network"
+2. **[OAuth 2](https://tools.ietf.org/html/rfc6749)** is an "authorization framework that enables applications to obtain limited access to user accounts on an HTTP service, such as Facebook, GitHub, and DigitalOcean." 
+3. **[SAML (Security Assertion Markup Language)](https://saml.xml.org/)** is an "XML-based, open-standard data format for exchanging authentication and authorization data between parties, in particular, between an identity provider and a service provider." SAML is often to provide Single Sign-On (SSO) to users, which allows to access different services using only one account.
```

Comment:
There is a grammar error in 
> SAML is often to provide Single Sign-On (SSO) to users, which allows to access different services using only one account.

You can try doing 
SAML is often provide to Single Sign-On (SSO) users, which allows access to different services using only one account.